### PR TITLE
rich-text formatting completed

### PR DESCRIPTION
**Pull request by iambenkay**
# What's New?
Includes a whole new way to format strings with various rich text identifiers.

```go
c := color.New()

c.PrintlnRT("This is the example of *Rich Text Formatting* in the ~terminal~. (fg:black|(bg:cyan|BLACK text on a CYAN background)).")
```
#### Output:
image

## Usage
BOLD - `*bold text*`
ITALICS - `_italicized text_`
UNDERLINE - `~underlined text~`
TEXT COLOR - `(fg:red|red text)`
BACKGROUND COLOR - `(bg:cyan|cyan background)`

Feel free to suggest anything you feel should be added to this PR.

### CODE REVIEW

**Comment by iambenkay** [comment]
This is a response to my issue #117

File: color.go [hunk 1]
```diff
@@ -44,6 +45,42 @@ type Attribute int
 
 const escape = "\x1b"
 
+// colorMap contains rich text identifier for color based formatting
+var colorMap = map[string]Attribute{
```
Comment (iambenkay) [inline color.go:49]: I am using this on line 552 in `wrapRichText` to dynamically assign the corresponding color to the symbol matched

**Comment by iambenkay** [comment]
@fatih Can you take a look at this.

**Comment by iambenkay** [comment]
@fatih @bogem @zddhub

**Review by zddhub — APPROVED** [review]
Look good to me!
Thanks for your contribution

**Pull request closed without merging**